### PR TITLE
docs: add withCustomMiddleware

### PR DESCRIPTION
**Pull request by SahilAujla**

### CODE REVIEW

File: site/packages/aa-core/provider/withCustomMiddleware.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Usage
+
+::: code-group
+
+```ts [example.ts]
+import { provider } from "./provider";
+
+// [!code focus:99]
+provider.withCustomMiddleware(CustomMiddlewareOverrideFunction);
```
Comment (avasisht23) [inline site/packages/aa-core/provider/withCustomMiddleware.md:27]: could you make the example more detailed? For instance, let's define `CustomMiddlewareOverrideFunction` and then maybe call buildUserOperation and show in a comment how resulting uo struct fields (changed in the resulting uo struct after build due to this middleware override

**Review by avasisht23 — COMMENTED** [review]
can you make the example more detailed? thanks!

File: site/packages/aa-core/provider/withCustomMiddleware.md (hunk 1 above)
Comment (SahilAujla) [reply site/packages/aa-core/provider/withCustomMiddleware.md:35]: @avasisht23 how does it look now?

File: site/packages/aa-core/provider/withCustomMiddleware.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - name: description
+      content: Overview of the withCustomMiddleware method and accessing the customMiddleware readonly field on ISmartAccountProvider
+  - - meta
+    - property: og:description
+      content: Overview of the withCustomMiddleware method and accessing the customMiddleware readonly field on ISmartAccountProvider
+---
+
+# withCustomMiddleware
+
+This method adds a final middleware step to the middleware stack that runs right before signature verification. It can be utilized when there's an RPC that fulfills most of the functionalities of the other middlewares. Instead of chaining multiple RPC calls via the default middlewares, you can delegate the work to that RPC.
```
[the remark this answers is not in the document]
Comment (SahilAujla) [reply site/packages/aa-core/provider/withCustomMiddleware.md]: Changed!

File: site/packages/aa-core/provider/withCustomMiddleware.md (hunk 1 above)
Comment (avasisht23) [reply site/packages/aa-core/provider/withCustomMiddleware.md:35]: use async / await syntax on the buildUserOperation. otw, lgtm

Comment (SahilAujla) [reply site/packages/aa-core/provider/withCustomMiddleware.md:35]: Done
image

File: packages/core/src/provider/types.ts [hunk 3]
```diff
@@ -263,9 +263,9 @@ export interface ISmartAccountProvider<
   withFeeDataGetter: (override: FeeDataMiddleware) => this;
 
   /**
-   * This adds a final middleware step to the middleware stack that runs right before signature verification.
-   * This can be used if you have an RPC that does most of the functions of the other middlewares for you and
-   * you want to delegate that work to that RPC instead of chaining together multiple RPC calls via the default middlwares.
+   * Adds a function to the end of the middleware call stack before signature verification.
```
Comment (moldy530) [inline packages/core/src/provider/types.ts:266]: actually it runs right before paymasterAndData middleware: https://github.com/OMGWINNING/aa-sdk-staging/pull/85/files#r1350480634 (or will)

Comment (SahilAujla) [reply packages/core/src/provider/types.ts:266]: @moldy530 do you have a preferred description here based on the above?

Comment (moldy530) [inline packages/core/src/provider/types.ts:266]:
```suggestion
   * Adds a function to the end of the middleware call stack before calling the paymaster middleware.
```

File: site/packages/aa-core/provider/withCustomMiddleware.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+// function to call buildUserOperation and log the modified callData
+async function displayModifiedCallData() {
+  const resultingUO = await provider.buildUserOperation(userOpData);
+  console.log("Modified callData:", resultingUO.callData);
+}
+
+// Call the function
+displayModifiedCallData();
```
Comment (moldy530) [inline site/packages/aa-core/provider/withCustomMiddleware.md:50]:
```suggestion
const resultingUO = await provider.buildUserOperation(userOpData);
console.log("Modified callData:", resultingUO.callData);
```

**Review by moldy530 — APPROVED** [review]

**Pull request merged**